WikiPreview.Fluent.Plugin/WikiPreviewSearchApp.cs

```diff
@@ -108,59 +111,84 @@ public ValueTask LoadSearchApplicationAsync()
-                        {
-                            bitmapImageResult = new BitmapImageResult(); // create empty if no image source.
-                        }
-
-                        WikiPreviewSearchResult wikiPreviewSearchResult = new()
-                        {
-                            Url = wikiUrl,
-                            PreviewImage = bitmapImageResult,
-                            DisplayedName = displayedName,
-                            ResultName = resultName,
-                            SearchedText = searchedText,
-                            Score = score,
-                            PageId = pageId,
-                            AdditionalInformation = additionalInfo
-                        };
-                        await channel.Writer.WriteAsync(wikiPreviewSearchResult).ConfigureAwait(false);
-                    }, cancellationToken).ContinueWith(_ => channel.Writer.Complete());
+                        }, cancellationToken)
+                        .ContinueWith(_ => channel.Writer.Complete(), CancellationToken.None);
                 else
                     channel.Writer.Complete();
-            });
+            }, CancellationToken.None);
 
             await foreach (WikiPreviewSearchResult item in channel.Reader.ReadAllAsync(cancellationToken))
                 yield return item;
         }
 
-        public ValueTask<ISearchResult> GetSearchResultForId(string serializedSearchObjectId)
+        public async ValueTask<WikiPreviewSearchResult> GetSearchResultForId(object serializedSearchObjectId)
         {
-            return new();
+            string pageId = serializedSearchObjectId as string;
+            if (string.IsNullOrWhiteSpace(pageId))
+                return default;
+
+            string url = "https://en.wikipedia.org/w/api.php?action=query&prop=extracts|pageimages&pageids=" +
+                         pageId +
+                         "&explaintext&exintro&pilicense=any&pithumbsize=100&format=json";
+
+            using var httpClient = new HttpClient();
+            var wiki = await httpClient.GetFromJsonAsync<Wiki>(url, _serializerOptions);
+            if (wiki == null) return default;
+
+            Dictionary<string, PageView>.ValueCollection pages = wiki.Query.Pages.Values;
+            if (pages is {Count: 0}) return default;
+
+            PageView pageView = pages.First();
+            WikiPreviewSearchResult wikiTask = await GenerateSearchResult(pageView, pageView?.Title);
+            return wikiTask;
+        }
+
+        private static async ValueTask<WikiPreviewSearchResult> GenerateSearchResult(PageView value,
+            string searchedText)
+        {
+            string resultName = value.Extract;
+            string displayedName = value.Title;
+            double score = displayedName.SearchDistanceScore(searchedText);
+            string pageId = value.PageId.ToString();
+            string wikiUrl = WikiRootUrl + displayedName.Replace(' ','_');
```

Comment:
You sure this is always a valid URL? Don't you get a valid ID/URL through the API call?
If not, you might want to use `HttpUtility.UrlEncode` instead of trying to fix the string yourself.

Reply:
I can't get the URL from API but yeah, it is always valid. I'll try the `HttpUtility.UrlEncode`. Thank you.

Reply:
The method `HttpUtility.UrlEncode` is only giving the undesired results. Since the URLs will always be valid and underscores give the original look of the Wikipedia URLs, I don't think there is any need for it.

Reply:
Did you use the encode only on the `displayedName`?

Reply:
Yeah, but it should only keep underscores where the `HttpUtility.UrlEncode` replacing spaces with plus (+) signs.